protoc-gen-gogo/generator/generator.go

```diff
@@ -1428,6 +1428,10 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
 		g.PrintComments(fmt.Sprintf("%s,%d,%d", enum.path, enumValuePath, i))
 
 		name := ccPrefix + *e.Name
+		if gogoproto.IsEnumCustomName(e) {
```

Comment:
The plugin should probably check that EnabledGoEnumPrefix is false if IsEnumCustomName is true.
If this is not the case the plugin should at least give a WARNING and possibly an error.
See the defaultcheck plugin for examples.